### ISSUE TITLE
OnPointerUp null reference

### ISSUE THREAD

**Issue by pkar70** [unoplatform/uno#2656]
## Current behavior
`System.NullReferenceException: 'Object reference not set to an instance of an object.'`
Method: `private bool OnPointerUp(PointerRoutedEventArgs args, bool isManagedBubblingEvent)`
Line 548: `if (_gestures.IsValueCreated)`
_gestures is null.

## Expected behavior
No error.

## How to reproduce it (as minimally and precisely as possible)
In my code:
MainPage.xaml.cs

     private async void uiBusStopList_DoubleTapped(object sender, Windows.UI.Xaml.Input.DoubleTappedRoutedEventArgs e)
    {
     string sMask = await DialogBoxInput("Enter Name");

...

         public async static System.Threading.Tasks.Task<string> DialogBoxInput(string sPrompt)
        {
            Windows.UI.Xaml.Controls.TextBox oInputTextBox = new Windows.UI.Xaml.Controls.TextBox();
            oInputTextBox.AcceptsReturn = false;
            Windows.UI.Xaml.Controls.ContentDialog oDlg = new Windows.UI.Xaml.Controls.ContentDialog();
            oDlg.Content = oInputTextBox;
            oDlg.PrimaryButtonText = "Go";
            oDlg.SecondaryButtonText = "Cancel";
            oDlg.Title = sPrompt;

            var oCmd = await oDlg.ShowAsync();
     #if !NETFX_CORE
            oDlg.Dispose();
     #endif

            if (oCmd != Windows.UI.Xaml.Controls.ContentDialogResult.Primary)
                return "";

            return oInputTextBox.Text;
        }

 

 When I remove `oDlg.Dispose()`, error disappears. 

## Environment

Nuget Package:

Package Version(s): 2.1.0-dev.1097
It works in 2.0.311-dev.3092

Affected platform(s):

- [ ] iOS
- [X] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [ ] 2017 (version: )
- [X] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

**Comment by dr1rrb** [comment]
This is because the `Dispose()` is inherited from the `Java.Lang.Object`, so here you are actually interfering with the Xamarin's marshaling system by disposing the managed representation of a native object. 

Then here when the native view receives a `MotionEvent` it's dynamically recreating a managed object to dispatch the event to. Doing this it will bypass the default constructor of the `UIElement` that's why the `_gestures` is `null`.

Long story short, you should never dispose a `UIElement`.

But I agree that it's a common issue with Xamarin where it dynamically recreates managed objects, and we should not fail for that, so I added a check to prevent this exception: https://github.com/unoplatform/uno/pull/2755

**Comment by dr1rrb** [comment]
Fixed by https://github.com/unoplatform/uno/pull/2755
